### ISSUE TITLE
list view show with enter a large folder and switch view mode slow with old version

### ISSUE THREAD

**Issue by ubuntuyin** [mate-desktop/caja#1586]
#### Expected behaviour
1、when i enter a folder which have a lot of files (20000 just in my computer and also use it) with list view，it will show all files after one minute just different icon view , when i use "sudo caja" in terminal, it will be fast just 3 seconds in list view. 

2、switch view mode is also slow in caja.

i know ubuntu 18.04 with caja is ok just with question 1，and i use this ubuntu version is many years, so i dont want to update it , and when i look at linux mint 18.1 with caja-1.16.6, the two session is also slow,so how can i fix it with little changes.

what the reason of the two question, i also wonder about this ~~~
 
#### MATE general version
mate-destktop 1.12.1-1

#### Package version
1.12.7-1ubuntu0.1

#### Linux Distribution
ubuntu 16.04

**Comment by ubuntuyin** [comment]
i dont find any bug report or patch for this(maybe i dont understand that), so i dont know how the question is fixed except update system and caja, so i ask here impudent.  ：）

**Comment by mbkma** [comment]
try `Edit->Preferences->Preview` and set all options to `never`. Is it faster?

**Comment by ubuntuyin** [comment]
thanks for your reply, and i try it but not effect,, list view must have to take one minutes for all files show. 
so when did it fixed, or never dealt with it, just updated dependencies, but I don't know which one dependency packages affected it in question 1（"sodu caja" is not a perfect modification scheme，question 2 is appearing in the latest version of caja）

**Comment by ubuntuyin** [comment]
i also report it at ubuntu:
https://answers.launchpad.net/ubuntu/+source/caja/+question/699711

**Comment by ubuntuyin** [comment]
i also report it at ubuntu:
https://answers.launchpad.net/ubuntu/+source/caja/+question/699711

**Comment by lukefromdc** [comment]
Most likely this comes from rendering so many icons and is inherited from Nautilus. Way back in 2008, the very pretty but heavy ubuntustudio-icon-theme had real problems rerendering icons on scroll in a large folder of audio files on then-old but still servicable Pentium 4 machines. All of the audio icons in that theme at that time were large .svg images, which made any issues with rendering the same icon many times over worse.  

To get around this we would have to do one of two things: Either render icons only for the files on display (thus bringing back or creating anew the scrolling issue I got back in 2008) or else for any repeatedly used mimetype store a bitmap of the decoded icon file in RAM at the exact pixel size it will be shown and reuse it as many times as needed. Not sure how much faster that would really be, might depend even on such things as using or not using a compositing window manager (without at lots of small writes to video RAM maybe?) Even this would not work if you had a large enough variety of mimetypes in one directory-or more likely a bunch of thumbnails.

 I've never, ever seen any Nautlius-based file manager open an icon view of 10,000+ files quickly.

**Comment by ubuntuyin** [comment]
i agree with you, now i wonder about is "sudo caja" will be ok,but why?
and question1 will be fast in many release version with caja higher version(>= ubuntu 18.04 / linux mint 18.3 / ubuntu mate 16.10)，i just think that caja maybe fix it, but i cant find the pathch for it , also maybe other dependence package(gail/mate-desktop/gtk/glib...) fix it for caja, i dont know much about it.
i want to fix it with lower version as a beginner，but I was lost~~~

**Comment by lukefromdc** [comment]
I have no idea what is going differently with sudo caja.

**Comment by lukefromdc** [comment]
Note that `sudo caja` is NOT faster on my setup with a big directory like /var/lib/dpkg/info

**Comment by ubuntuyin** [comment]
i test i with many version installed with release version(<= ubuntu 18.04 / linux mint 18.3 / ubuntu mate 16.10),so what version did you have?
Test ubunt 1604 with "Files"(nautiul) is faster too.

**Comment by lukefromdc** [comment]
I've got caja 1.26 from current git master locally compiled (as is the rest of MATE) over Debian Unstable

**Comment by ubuntuyin** [comment]
your version of caja is ok, i had tested already,now the question is old version of caja,
Match the version：
ubuntu 18.04-> caja 1.20.2
linux mint 18.3-> caja 1.18.5
ubuntu mate-> caja 1.16.0
so ,if you want to get lower session,you should use caja lower version before the version above.
and you can also use caja 1.12.7,it always slow in many ubuntu release version.

**Comment by ubuntuyin** [comment]
and i dont know how and when caja fix it , i cant update my pc system and caja version directly just for skip this question。

**Comment by mbkma** [comment]
So do you say that this issue is fixed in Caja 1.26?

**Comment by ubuntuyin** [comment]
> So do you say that this issue is fixed in Caja 1.26?

yes, i said it above already. Now i don't know how it work ok with caja's higher version since i didn't find that patch which fix question 1; maybe caja did nothing for it , emmm.
and question 2 is also exist in higher version, you know.
